File: backend/app/models/onboarding.py
from pydantic import BaseModel, EmailStr
from typing import Optional
from datetime import datetime


class OnboardingFormData(BaseModel):
    """Complete onboarding form data"""
    # Step 1: Company Information
    company_name: str
    company_domain: str
    company_size: str
    
    # Step 2: Industry Information
    company_industry: str
    
    # Step 3: Company Roles
    company_roles: str
    
    # Step 4: Your Needs
    your_needs: str


class OnboardingRequest(BaseModel):
    """Request model for onboarding submission"""
    user_id: Optional[str] = None  # Will be set from JWT token
    form_data: OnboardingFormData


class OnboardingResponse(BaseModel):
    """Response model for onboarding submission"""
    success: bool
    message: str
    onboarding_id: Optional[str] = None
    workspace_created: bool = False


class OnboardingError(BaseModel):
    """Error model for onboarding"""
    error: str
    message: str
    details: Optional[dict] = None


class OnboardingStatusResponse(BaseModel):
    """Response model for onboarding status/data retrieval"""
    user_id: str
    onboarding_completed: bool
    workspace_created: bool
    onboarding_id: Optional[str] = None
    company_name: Optional[str] = None
    company_domain: Optional[str] = None
    company_size: Optional[str] = None
    company_industry: Optional[str] = None
    company_roles: Optional[str] = None
    your_needs: Optional[str] = None
    last_updated: Optional[str] = None

File: frontend/src/types/onboarding.ts
// This is the response for the GET onboarding
export interface Onboarding {
  user_id: string
  onboarding_completed: boolean
  workspace_created: boolean
  onboarding_id: string
  company_name: string
  company_domain: string
  company_size: string
  company_industry: string
  company_roles: string
  your_needs: string
  last_updated: string
}

// This is the request for the POST onboarding
export type OnboardingRequest = Omit<
  Onboarding,
  | 'user_id'
  | 'onboarding_completed'
  | 'workspace_created'
  | 'last_updated'
  | 'onboarding_id'
>

// This is the response for the POST onboarding
export interface OnboardingResponse {
  success: boolean
  message: string
  onboarding_id: number
  workspace_created: boolean
  company_domain: string
  full_domain: string
}

// Company size options for validation
export type CompanySize =
  | '1-10'
  | '11-50'
  | '51-100'
  | '101-200'
  | '201-500'
  | '500+'

// Frontend form data interface (camelCase for React components)
export interface OnboardingFormData {
  companyName: string
  companyDomain: string
  companySize: CompanySize
  companyIndustry: string
  companyRoles: string
  yourNeeds: string
}

// Domain availability check response
export interface DomainAvailabilityResponse {
  domain: string
  available: boolean
  full_domain: string
  message: string
}
